iotedgedev/azurecli.py

```diff
@@ -0,0 +1,265 @@
+            return az_args
+        return [" ".join(az_args)] 
+
+    def invoke_az_cli_outproc(self, args, error_message=None, stdout_io = None, stderr_io = None):
+        try:
+            if stdout_io or stderr_io:
+                process = subprocess.Popen(self.prepare_az_cli_args(args), stdout=subprocess.PIPE, stderr=subprocess.PIPE, shell=True)
+                stdout_data, stderr_data = process.communicate()
+                if stdout_io and stdout_data != "":
+                    stdout_io.writelines(self.decode(stdout_data))
+                if stderr_io and stderr_data != "":
+                    stderr_io.writelines(self.decode(stderr_data))
+            else:
+                process = subprocess.Popen(self.prepare_az_cli_args(args), shell=True)
+                process.communicate()
+            
+            if process.returncode != 0:
+                if error_message:
+                    self.output.error(error_message)
+                return False
+
+        except Exception as e:
+            if error_message:
+                self.output.error(error_message)
+            self.output.error(str(e))
+            return False
+
+        self.output.footer("Success")
+
+        return True
+
+    def invoke_az_cli(self, args, error_message=None, stdout_io=None):
+        try:
+            exit_code = self.az_cli.invoke(args, out_file=stdout_io)
+            if exit_code and exit_code != 0:
+                if error_message:
+                    self.output.error(error_message)
+                return False
+        except Exception as e:
+            if error_message:
+                self.output.error(error_message)
+            self.output.error(str(e))
+            return False
+
+        self.output.footer("Success")
+
+        return True
+
+    def add_extension(self, name):
+        self.output.header(f("Adding extension {name}"))
+
+        return self.invoke_az_cli_outproc(["extension", "add", "--name", name,
+                                   "--yes"],
+                                  f("Error while adding extension {name}."))
+
+    def extension_exists(self, name):
+        self.output.header(f("Checking for extension {name}"))
+
+        return self.invoke_az_cli_outproc(["extension", "show", "--name", name, "--output", "table"],
+                                  f("Error while checking for extension {name}."))
+
+    def user_has_logged_in(self):
+        self.output.header("Checking for cached credentials.")
+
+        
+        with output_io_cls() as io:
+            result = self.invoke_az_cli_outproc(["account", "show"], stdout_io =io)
+
+            if result:
+                try:
+                    out_string = io.getvalue()
+                    data = json.loads(out_string)
+                    return data["id"]
+                except Exception:
+                    pass
+        self.output.info("No credentials in cache.")
```

Comment:
Change to "Azure CLI credentials not found. Please follow instructions below to login to the Azure CLI."

Reply:
I would like to see more space between these commands...and across the board I feel like it a crowded output
![image](https://user-images.githubusercontent.com/2163001/36031514-d89c8ed2-0d5f-11e8-98fe-c6fb1e55ec82.png)


What do you think about doing something like this for "header"

============================
====== LISTING SUBSCRIPTIONS =====
============================